components/EventBanner/EventBanner.tsx

```diff
@@ -0,0 +1,79 @@
+import { getParsedDate } from "utils/date";
+import CalendarTrans from "./assets/calendar-trans.svg?react";
+import ArrowRight from "./assets/arrow-right.svg?react";
+import MapPin from "./assets/map-pin.svg?react";
+import Link from "next/link";
+import styles from "./EventBanner.module.css";
+
+export interface EventProps {
+  end?: string | null;
+  link: string;
+  location: string;
+  start: string;
+  title: string;
+  featured?: boolean;
+}
+
+export interface Events {
+  events: EventProps[];
+}
+
+export interface Event {
+  selectedEvent: EventProps;
+}
+
+const getComingEvent = (events: EventProps[]) => {
+  const currentDate = new Date();
+  let selectedEvent = null;
+  for (const event of events) {
+    // If the event is not featured, skip it
+    if (!event?.featured) {
+      continue;
+    }
+
+    const startDate = new Date(event.start);
+    const endDate = event.end ? new Date(event.end) : startDate;
+
+    if (currentDate <= endDate) {
+      selectedEvent = event;
+      break;
+    }
+  }
```

Comment:
consider using `Array.find()` here with the specified criteria of `event.featured` and the date comparison. This would simplify the function and be a bit more readble. If `find` doesn't find anything, it returns `undefined` which can be used in the same fashion in your `EventBanner`